sync/src/types.rs

```diff
@@ -308,65 +386,26 @@ impl DownloadScheduler {
         self.task_count.saturating_sub(self.hashes.len())
     }
 
-    pub(crate) fn task_count(&self) -> usize {
+    pub(crate) const fn task_count(&self) -> usize {
         self.task_count
     }
 
-    fn adjust(&mut self, time: u64, len: u64) {
-        let now = unix_time_as_millis();
-        // 8 means default max outbound
-        // All synchronization tests are based on the assumption of 8 nodes.
-        // If the number of nodes is increased, the number of requests and processing time will increase,
-        // and the corresponding adjustment criteria are needed, so the adjustment is based on 8.
-        //
-        // note: This is an interim scenario and the next step is to consider the median response time
-        // within a certain interval as the adjustment criterion
-        let (quotient, remainder) = if len > 8 { (len >> 3, len % 8) } else { (1, 0) };
-        // Dynamically adjust download tasks based on response time.
-        //
-        // Block max size about 700k, Under 10m/s bandwidth it may cost 1s to response
-        // But the mark is on the same time, so multiple tasks may be affected by one task,
-        // causing all responses to be in the range of reduced tasks.
-        // So the adjustment to reduce the number of tasks will be calculated by modulo 3 with the actual number of triggers.
-        // Adjust for each block received.
-        match ((now - time) / quotient).saturating_sub(remainder * 100) {
-            // Within 500ms of response, considered better to communicate with that node's network
-            0..=500 => {
-                if self.task_count < FIRST_LEVEL_MAX - 1 {
-                    self.task_count += 2
-                } else {
-                    self.breakthroughs_count += 1
-                }
-            }
-            // Within 1000ms of response time, the network is relatively good and communication should be maintained
-            501..=1000 => {
-                if self.task_count < FIRST_LEVEL_MAX {
-                    self.task_count += 1
-                } else {
-                    self.breakthroughs_count += 1
-                }
-            }
-            // Response time within 1500ms, acceptable range, but not relatively good for nodes and do not adjust
-            1001..=1500 => (),
-            // Above 1500ms, relatively poor network, reduced
-            _ => {
-                self.timeout_count += 1;
-                if self.timeout_count > 3 {
-                    self.task_count = self.task_count.saturating_sub(1);
-                    self.timeout_count = 0;
-                }
-            }
+    fn increase(&mut self, num: usize) {
+        if self.task_count < MAX_BLOCKS_IN_TRANSIT_PER_PEER {
```

Comment:
It's wrong when `num > 1`.

Reply:
Nothing wrong, it is allowed to be greater than `MAX_BLOCKS_IN_TRANSIT_PER_PEER`, only need to control not to rise indefinitely

Reply:
This issue is that the upper is checked before adding `num` to it. So if a node may have a higher task count when adding 1 then adding 2. It's not issue since we only use 1 and 2 now. But later if we decide to use a higher `num`, it can cause strange result.

Reply:
I don't think it will be a problem, but I'll change it if you insist.